lib/openstax/exchange/client/real_client.rb

```diff
@@ -0,0 +1,80 @@
+require 'oauth2'
+
+module OpenStax
+  module Exchange
+    module Client
+      class RealClient
+        include ClientInstance
```

Comment:
I guess my issue with `ClientInstance` is that it contains two methods, one of which is only used in the specs and the other of which doesn't really appear to be a public method (`token` is an implementation detail).  So I would argue that those methods should not be there and that the resulting empty class is unnecessary.  

Also, it sounds like you're trying to enforce that the clients follow an interface.  Since Ruby doesn't have compile-time type verification, the best we can really do is have the specs check for expected methods (which can happen without including abstract methods that raise exceptions).